stdlib/hashtbl.ml

```diff
@@ -81,11 +81,11 @@ let create ?(random = !randomized) initial_size =
   { initial_size = s; size = 0; seed = seed; data = Array.make s Empty }
 
 let clear h =
-  h.size <- 0;
-  let len = Array.length h.data in
-  for i = 0 to len - 1 do
-    h.data.(i) <- Empty
-  done
+  if h.size > 0 then begin
+    h.size <- 0;
+    let len = Array.length h.data in
+    Array.fill h.data 0 len Empty
```

Comment:
Why not simply `Array.fill h.data 0 (Array.length h.data) Empty` ?

Reply:
I just kept that from the original code.  I've now added a commit following your suggestion.